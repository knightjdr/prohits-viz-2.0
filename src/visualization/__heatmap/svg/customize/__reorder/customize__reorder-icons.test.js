import React from 'react';
import { shallow } from 'enzyme';

import reorderIcons from './customize__reorder-icons';

const mouseDown = jest.fn();

describe('Reorder icons', () => {
  describe('for columns', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(
        <div>
          { reorderIcons(5, 20, 'col', mouseDown) }
        </div>,
      );
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should have five g elements', () => {
      expect(wrapper.find('g').length).toBe(5);
    });

    describe('mouse events', () => {
      let g;

      beforeAll(() => {
        g = wrapper.find('g').first();
      });

      it('should call mouseDown', () => {
        mouseDown.mockClear();
        g.simulate('mousedown', {});
        expect(mouseDown).toHaveBeenCalledWith({}, 0);
      });
    });
  });

  describe('for rows', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(
        <div>
          { reorderIcons(5, 20, 'row', mouseDown) }
        </div>,
      );
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should have five g elements', () => {
      expect(wrapper.find('g').length).toBe(5);
    });

    describe('mouse events', () => {
      let g;

      beforeAll(() => {
        g = wrapper.find('g').first();
      });

      it('should call mouseDown', () => {
        mouseDown.mockClear();
        g.simulate('mousedown', {});
        expect(mouseDown).toHaveBeenCalledWith({}, 0);
      });
    });
  });
});
