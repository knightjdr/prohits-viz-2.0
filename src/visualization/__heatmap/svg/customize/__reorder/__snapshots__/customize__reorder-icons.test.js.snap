// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reorder icons for columns should match snapshot 1`] = `
<div>
  <g
    className="delete-icon__circle"
    cursor="pointer"
    key="icon-0"
    onMouseDown={[Function]}
    transform="
            translate(2 0)
            scale(1.6)
          "
  >
    <circle
      cx="5"
      cy="5"
      fill="#336B87"
      r="5"
    />
    <path
      d="M3.333,6.176V5.546h3.334v0.629c0,0.293,0.354,0.439,0.56,0.232l1.178-1.176c0.127-0.128,0.127-0.336,0-0.464 L7.227,3.591c-0.206-0.207-0.56-0.061-0.56,0.231v0.63H3.333v-0.63c0-0.292-0.353-0.438-0.56-0.231L1.596,4.768 c-0.128,0.128-0.128,0.336,0,0.464l1.178,1.176C2.98,6.615,3.333,6.469,3.333,6.176z"
      fill="#fff"
    />
  </g>
  <g
    className="delete-icon__circle"
    cursor="pointer"
    key="icon-1"
    onMouseDown={[Function]}
    transform="
            translate(22 0)
            scale(1.6)
          "
  >
    <circle
      cx="5"
      cy="5"
      fill="#336B87"
      r="5"
    />
    <path
      d="M3.333,6.176V5.546h3.334v0.629c0,0.293,0.354,0.439,0.56,0.232l1.178-1.176c0.127-0.128,0.127-0.336,0-0.464 L7.227,3.591c-0.206-0.207-0.56-0.061-0.56,0.231v0.63H3.333v-0.63c0-0.292-0.353-0.438-0.56-0.231L1.596,4.768 c-0.128,0.128-0.128,0.336,0,0.464l1.178,1.176C2.98,6.615,3.333,6.469,3.333,6.176z"
      fill="#fff"
    />
  </g>
  <g
    className="delete-icon__circle"
    cursor="pointer"
    key="icon-2"
    onMouseDown={[Function]}
    transform="
            translate(42 0)
            scale(1.6)
          "
  >
    <circle
      cx="5"
      cy="5"
      fill="#336B87"
      r="5"
    />
    <path
      d="M3.333,6.176V5.546h3.334v0.629c0,0.293,0.354,0.439,0.56,0.232l1.178-1.176c0.127-0.128,0.127-0.336,0-0.464 L7.227,3.591c-0.206-0.207-0.56-0.061-0.56,0.231v0.63H3.333v-0.63c0-0.292-0.353-0.438-0.56-0.231L1.596,4.768 c-0.128,0.128-0.128,0.336,0,0.464l1.178,1.176C2.98,6.615,3.333,6.469,3.333,6.176z"
      fill="#fff"
    />
  </g>
  <g
    className="delete-icon__circle"
    cursor="pointer"
    key="icon-3"
    onMouseDown={[Function]}
    transform="
            translate(62 0)
            scale(1.6)
          "
  >
    <circle
      cx="5"
      cy="5"
      fill="#336B87"
      r="5"
    />
    <path
      d="M3.333,6.176V5.546h3.334v0.629c0,0.293,0.354,0.439,0.56,0.232l1.178-1.176c0.127-0.128,0.127-0.336,0-0.464 L7.227,3.591c-0.206-0.207-0.56-0.061-0.56,0.231v0.63H3.333v-0.63c0-0.292-0.353-0.438-0.56-0.231L1.596,4.768 c-0.128,0.128-0.128,0.336,0,0.464l1.178,1.176C2.98,6.615,3.333,6.469,3.333,6.176z"
      fill="#fff"
    />
  </g>
  <g
    className="delete-icon__circle"
    cursor="pointer"
    key="icon-4"
    onMouseDown={[Function]}
    transform="
            translate(82 0)
            scale(1.6)
          "
  >
    <circle
      cx="5"
      cy="5"
      fill="#336B87"
      r="5"
    />
    <path
      d="M3.333,6.176V5.546h3.334v0.629c0,0.293,0.354,0.439,0.56,0.232l1.178-1.176c0.127-0.128,0.127-0.336,0-0.464 L7.227,3.591c-0.206-0.207-0.56-0.061-0.56,0.231v0.63H3.333v-0.63c0-0.292-0.353-0.438-0.56-0.231L1.596,4.768 c-0.128,0.128-0.128,0.336,0,0.464l1.178,1.176C2.98,6.615,3.333,6.469,3.333,6.176z"
      fill="#fff"
    />
  </g>
</div>
`;

exports[`Reorder icons for rows should match snapshot 1`] = `
<div>
  <g
    className="delete-icon__circle"
    cursor="pointer"
    key="icon-0"
    onMouseDown={[Function]}
    transform="
            translate(0 2)
            scale(1.6)
          "
  >
    <circle
      cx="5"
      cy="5"
      fill="#336B87"
      r="5"
    />
    <path
      d="M6.177,6.667h-0.63V3.333h0.63c0.292,0,0.439-0.354,0.231-0.56L5.232,1.596c-0.128-0.128-0.336-0.128-0.464,0 L3.592,2.773c-0.207,0.207-0.061,0.56,0.231,0.56h0.63v3.334h-0.63c-0.292,0-0.438,0.353-0.231,0.56l1.176,1.178 c0.128,0.128,0.336,0.128,0.464,0l1.176-1.178C6.616,7.02,6.469,6.667,6.177,6.667z"
      fill="#fff"
    />
  </g>
  <g
    className="delete-icon__circle"
    cursor="pointer"
    key="icon-1"
    onMouseDown={[Function]}
    transform="
            translate(0 22)
            scale(1.6)
          "
  >
    <circle
      cx="5"
      cy="5"
      fill="#336B87"
      r="5"
    />
    <path
      d="M6.177,6.667h-0.63V3.333h0.63c0.292,0,0.439-0.354,0.231-0.56L5.232,1.596c-0.128-0.128-0.336-0.128-0.464,0 L3.592,2.773c-0.207,0.207-0.061,0.56,0.231,0.56h0.63v3.334h-0.63c-0.292,0-0.438,0.353-0.231,0.56l1.176,1.178 c0.128,0.128,0.336,0.128,0.464,0l1.176-1.178C6.616,7.02,6.469,6.667,6.177,6.667z"
      fill="#fff"
    />
  </g>
  <g
    className="delete-icon__circle"
    cursor="pointer"
    key="icon-2"
    onMouseDown={[Function]}
    transform="
            translate(0 42)
            scale(1.6)
          "
  >
    <circle
      cx="5"
      cy="5"
      fill="#336B87"
      r="5"
    />
    <path
      d="M6.177,6.667h-0.63V3.333h0.63c0.292,0,0.439-0.354,0.231-0.56L5.232,1.596c-0.128-0.128-0.336-0.128-0.464,0 L3.592,2.773c-0.207,0.207-0.061,0.56,0.231,0.56h0.63v3.334h-0.63c-0.292,0-0.438,0.353-0.231,0.56l1.176,1.178 c0.128,0.128,0.336,0.128,0.464,0l1.176-1.178C6.616,7.02,6.469,6.667,6.177,6.667z"
      fill="#fff"
    />
  </g>
  <g
    className="delete-icon__circle"
    cursor="pointer"
    key="icon-3"
    onMouseDown={[Function]}
    transform="
            translate(0 62)
            scale(1.6)
          "
  >
    <circle
      cx="5"
      cy="5"
      fill="#336B87"
      r="5"
    />
    <path
      d="M6.177,6.667h-0.63V3.333h0.63c0.292,0,0.439-0.354,0.231-0.56L5.232,1.596c-0.128-0.128-0.336-0.128-0.464,0 L3.592,2.773c-0.207,0.207-0.061,0.56,0.231,0.56h0.63v3.334h-0.63c-0.292,0-0.438,0.353-0.231,0.56l1.176,1.178 c0.128,0.128,0.336,0.128,0.464,0l1.176-1.178C6.616,7.02,6.469,6.667,6.177,6.667z"
      fill="#fff"
    />
  </g>
  <g
    className="delete-icon__circle"
    cursor="pointer"
    key="icon-4"
    onMouseDown={[Function]}
    transform="
            translate(0 82)
            scale(1.6)
          "
  >
    <circle
      cx="5"
      cy="5"
      fill="#336B87"
      r="5"
    />
    <path
      d="M6.177,6.667h-0.63V3.333h0.63c0.292,0,0.439-0.354,0.231-0.56L5.232,1.596c-0.128-0.128-0.336-0.128-0.464,0 L3.592,2.773c-0.207,0.207-0.061,0.56,0.231,0.56h0.63v3.334h-0.63c-0.292,0-0.438,0.353-0.231,0.56l1.176,1.178 c0.128,0.128,0.336,0.128,0.464,0l1.176-1.178C6.616,7.02,6.469,6.667,6.177,6.667z"
      fill="#fff"
    />
  </g>
</div>
`;
